src/pages/settings/Wallet/ExpensifyCardPage.tsx

```diff
@@ -16,158 +16,92 @@ import useNetwork from '@hooks/useNetwork';
+    draftValues: OnyxEntry<GetPhysicalCardForm>;
 
-    /** Navigation route context info provided by react navigation */
-    route: PropTypes.shape({
-        params: PropTypes.shape({
-            /** domain passed via route /settings/wallet/card/:domain */
-            domain: PropTypes.string,
-        }),
-    }).isRequired,
+    /** Login info */
+    loginList: OnyxEntry<LoginList>;
 };
 
-const defaultProps = {
-    cardList: null,
-    draftValues: {
-        addressLine1: '',
-        addressLine2: '',
-        city: '',
-        state: '',
-        country: '',
-        zipPostCode: '',
-        phoneNumber: '',
-        legalFirstName: '',
-        legalLastName: '',
-    },
-    loginList: {},
-    privatePersonalDetails: {
-        legalFirstName: '',
-        legalLastName: '',
-        phoneNumber: null,
-        address: {
-            street: '',
-            city: '',
-            state: '',
-            zip: '',
-            country: '',
-        },
-    },
-};
+type ExpensifyCardPageProps = ExpensifyCardPageOnyxProps & StackScreenProps<PublicScreensParamList, typeof SCREENS.TRANSITION_BETWEEN_APPS>;
 
 function ExpensifyCardPage({
     cardList,
     draftValues,
-    loginList,
-    privatePersonalDetails,
+    loginList = {},
+    privatePersonalDetails = {},
     route: {
-        params: {domain},
+        params: {domain = ''},
     },
-}) {
+}: ExpensifyCardPageProps) {
     const styles = useThemeStyles();
     const {isOffline} = useNetwork();
     const {translate} = useLocalize();
     const domainCards = useMemo(() => cardList && CardUtils.getDomainCards(cardList)[domain], [cardList, domain]);
-    const virtualCard = useMemo(() => (domainCards && _.find(domainCards, (card) => card.isVirtual)) || {}, [domainCards]);
-    const physicalCard = useMemo(() => (domainCards && _.find(domainCards, (card) => !card.isVirtual)) || {}, [domainCards]);
+    const virtualCard = useMemo(() => domainCards?.find((card) => card.isVirtual), [domainCards]);
+    const physicalCard = useMemo(() => domainCards?.find((card) => !card.isVirtual), [domainCards]);
 
     const [isLoading, setIsLoading] = useState(false);
     const [isNotFound, setIsNotFound] = useState(false);
-    const [details, setDetails] = useState({});
+    const [details, setDetails] = useState<TCardDetails>();
     const [cardDetailsError, setCardDetailsError] = useState('');
 
     useEffect(() => {
         if (!cardList) {
             return;
         }
-        setIsNotFound(_.isEmpty(virtualCard) && _.isEmpty(physicalCard));
+        setIsNotFound(isEmptyObject(virtualCard) && isEmptyObject(physicalCard));
     }, [cardList, physicalCard, virtualCard]);
 
-    const formattedAvailableSpendAmount = CurrencyUtils.convertToDisplayString(physicalCard.availableSpend || virtualCard.availableSpend || 0);
+    const formattedAvailableSpendAmount = CurrencyUtils.convertToDisplayString(physicalCard?.availableSpend ?? virtualCard?.availableSpend ?? 0);
```

Comment:
This will cause regression if `physicalCard.availableSpend` = 0 and `virtualCard.availableSpend > 0`